tools/pastel-utility/cmd/init.go

```diff
@@ -0,0 +1,307 @@
+			resp.Body.Close()
+		} else {
+			log.WithContext(ctx).Infof("zksnark file %s already exists and force set to %v", zksnarkParamsPath, force)
+		}
+	}
+	log.WithContext(ctx).Info("ZkSnark params download completed.\n")
+	return nil
+}
+
+// checkLocalAndRouterFirewalls checks local and router firewalls and suggest what to open
+func checkLocalAndRouterFirewalls(ctx context.Context, requiredPorts []string) error {
+	baseURL := "https://portchecker.com?q=" + strings.Join(requiredPorts[:], ",")
+	// resp, err := http.Get(baseURL)
+	// if err != nil {
+	// 	log.WithContext(ctx).WithError(err).Errorf("Error requesting url\n")
+	// 	return errors.Errorf("Failed to request port url %v \n", err)
+	// }
+	// defer resp.Body.Close()
+	// ok := resp.StatusCode == http.StatusOK
+	ok := true
+	if ok {
+		log.WithContext(ctx).Info("Your ports {} are opened and can be accessed by other PAstel nodes! ", baseURL)
+	} else {
+		log.WithContext(ctx).Info("Your ports {} are NOT opened and can NOT be accessed by other Pastel nodes!\n Please open this ports in your router firewall and in {} firewall", baseURL, "func_to_check_OS()")
+	}
+	return nil
+}
+
+// runWalletSubCommand runs wallet subcommand
+func runWalletSubCommand(ctx context.Context, config *configs.Config) error {
+	log.WithContext(ctx).Infoln("Wallet Node Start")
+	defer log.WithContext(ctx).Infoln("Wallet Node End")
+
+	log.WithContext(ctx).Infof("Config: ")
+	log.WithContext(ctx).Infof(config.String())
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	sys.RegisterInterruptHandler(cancel, func() {
+		log.WithContext(ctx).Info("Interrupt signal received. Gracefully shutting down...")
+		os.Exit(0)
+	})
+
+	walletNodePath := filepath.Join(config.WorkDir, "walletnode")
+
+	if _, err := os.Stat(walletNodePath); os.IsNotExist(err) || config.Init.Force {
+		if err := os.MkdirAll(walletNodePath, os.ModePerm); err != nil {
+			return errors.Errorf("could not create walletnode path %s, %v", walletNodePath, err)
+		}
+	}
+
+	v := viper.New()
+	v.SetConfigType("json")
+	if err := v.ReadConfig(bytes.NewBufferString(config.Init.String())); err != nil {
+		return err
+	}
+
+	walletConfigPath := fmt.Sprintf("%s/wallet.env", walletNodePath)
+
+	if config.Init.Force {
+		if err := v.WriteConfigAs(walletConfigPath); err != nil {
+			return err
+		}
+	} else {
+		if err := v.SafeWriteConfigAs(walletConfigPath); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// runSuperNodeSubCommand runs wallet subcommand
+func runSuperNodeSubCommand(ctx context.Context, config *configs.Config) error {
+	log.WithContext(ctx).Infoln("Super Node Start")
+	defer log.WithContext(ctx).Infoln("Super Node End")
+
+	log.WithContext(ctx).Infof("Config: ")
+	log.WithContext(ctx).Infof(config.String())
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	sys.RegisterInterruptHandler(cancel, func() {
+		log.WithContext(ctx).Info("Interrupt signal received. Gracefully shutting down...")
+		os.Exit(0)
```

Comment:
Again, It looks like this was done on purpose.
Have you ever heard about `<-ctx.Done()` ?